lib/platform/linux/platform.c

```diff
@@ -170,10 +178,30 @@ static void * poll_for_indication(void * unused)
     // Initially wait for 500ms before any polling
     uint32_t wait_before_next_polling_ms = 500;
 
-    while (m_polling_thread_running)
+    m_polling_thread_state_request = POLLING_THREAD_RUN;
+
+     while (m_polling_thread_state_request != POLLING_THREAD_STOP)
     {
         usleep(wait_before_next_polling_ms * 1000);
 
+        if (!m_queue_empty)
+        {
+            // Dispatch did not process all indications, just wait for it to
+            wait_before_next_polling_ms = POLLING_INTERVAL_MS;
+            continue;
+        }
+
+        if(m_polling_thread_state_request == POLLING_THREAD_STOP_REQUESTED)
```

Comment:
Shouldn't we add a timeout here, to be sure close always ends in a given time.
We could start with a hardcoded 60sec and then add it as parameter to the platform api.

So ensuring that we never stay more than 60s in POLLING_THREAD_STOP_REQUESTED state.